### PR TITLE
feat: add segment descriptions to Campaigns and Segments tabs

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Makes the descriptions for segment action cards in the Segments tab (and segment container headers in the Campaigns tab) more useful. Instead of showing the creation date, it now summarizes the configuration for each segment. This could get slightly redundant if the segments are named after the intended use, but I find it really useful to be able to see at a glance what each segment is without having to visit the edit screen.

Segments tab:

Screen Shot 2021-02-03 at 2 07 00 PM

Campaigns tab:

Screen Shot 2021-02-03 at 2 05 49 PM

Also creates a link from the segments' names in the Campaigns tab to each segment's edit page, for added convenience.

~Question: I tried to make the descriptions as terse as possible so as not to visually muddy the UI, but in some cases the phrasing differs from the option labels—should we ensure that the option names match?~ I think we should, and I implemented as a draft in #848.

### How to test the changes in this Pull Request:

1. Check out this branch, run `npm run build`
2. Visit the Campaigns and Segments tabs.
3. In both tabs, verify that each segment is accompanied by a short summary of its configuration.
4. In the Campaigns tab, verify that clicking on the segment name directs you to that segment's edit page.
5. Create new segments and edit existing segments, using a mix of every available configuration option in some segments and leaving some fields blank in others. Verify that the descriptions accurately portray the configuration for each segment, showing only the options that differ from the default.

Note that for segments with Category Affinity set, it will take a second or two for the category names to appear in the description. This is because the configuration saves category IDs only, so we need to fetch the names via REST.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/popups/views/segments/SegmentsList.js [hunk 1]
```diff
@@ -131,6 +132,14 @@ const SegmentActionCard = ( {
 		resortSegments( target );
 	};
 
+	if ( 0 < segment.configuration?.favorite_categories?.length ) {
+		getFavoriteCategoryNames(
```
Comment (adekbadek) [inline assets/wizards/popups/views/segments/SegmentsList.js:136]: Here and in `SegmentGroup`, this function runs on each render, resulting in a ton of unnecessary requests. These calls should be wrapped in `useEffect` hooks.

**Review by adekbadek — CHANGES_REQUESTED** [review]

File: assets/wizards/popups/views/segments/SegmentsList.js (hunk 1 above)
Comment (dkoo) [reply assets/wizards/popups/views/segments/SegmentsList.js]: Good call—updated in 549fc70.

**Review by adekbadek — CHANGES_REQUESTED** [review]
For a segment with multiple favorite categories set, I can only see one in the segment description. After each REST request to the category endpoint, it's name is updated.

**Comment by dkoo** [comment]
> For a segment with multiple favorite categories set, I can only see one in the segment description. After each REST request to the category endpoint, it's name is updated.

Fixed in 0e251a9

**Review by adekbadek — APPROVED** [review]

**Pull request merged**